pkg/handler/enqueue_owner.go

```diff
@@ -197,3 +201,49 @@ func (e *enqueueRequestForOwner) getOwnersReferences(object metav1.Object) []met
 	// No Controller OwnerReference found
 	return nil
 }
+
+// logEvent logs an event with the version, kind and name of the object and its owner.
+func (e *enqueueRequestForOwner) logEvent(eventType string, object, newObject client.Object) {
```

Comment:
I am personally not a big fan of func signatures like this one - with positional parameters. Could using the builder pattern be an option here?

Reply:
Agreed. But a builder pattern just for `logEvent` could be an overkill just for one helper. I'll rework on this bit, to fetch the owner references from both the objects and then pass it to logevent so that its a bit more cleaner.